helm/cluster-autoscaler-app/values.yaml

```diff
@@ -63,6 +63,16 @@ balancingIgnoreLabels:
   balancing-ignore-label_2: vpc.amazonaws.com/eniConfig
   balancing-ignore-label_3: giantswarm.io/machine-deployment
 
+# Node-level settings for the deployment
+node:
+  # sets spec.template.spec.nodeSelector for the deployment
+  # The default value should be ok for most cases running on control plane nodes (it gets translated to node-role.kubernetes.io/master for k8s <v1.24.0), managed k8s services (e.g. EKS) need customization.
+  # Set it to {} to remove the nodeSelector.
```

Comment:
Have you actually tried this?

Spoiler: It doesn't. As far as I remember values specified on chart installation merge with the existing objects and therefore retain the defaults. This only works for arrays and setting `[]`.

Reply:
I tested it with helm template; if it is a new installation where we don't want to add a nodeSelector (e.g., on EKS), a nodeSelector never gets added to the deployment, which is what we want to achieve with this (to allow it to run on worker nodes).

Reply:
Yes, I get that. But this is the default value by now and you can not remove it by simply setting...

```yaml
node:
  nodeSelector: {}
```

... to remove them in the app's user values. So your docs are wrong. It should be `nodeSelector: null`.

Reply:
The line 26 in deployment.yaml (`{{- if ((.Values.node).nodeSelector) }}`) effectively removes the nodeSelector from the rendered template if the object is empty.

Setting `nodeSelector: null` instead does make the template fail the validation, because nodeSelector in the schema is defined as an object:
```
Error: values don't meet the specifications of the schema(s) in the following chart(s):
cluster-autoscaler-app:
- node.nodeSelector: Invalid type. Expected: object, given: null
```